docs/platform-specification.md

```diff
@@ -768,6 +768,60 @@ All the tools launched to compile or upload a sketch will have the following env
 contain multiple space-delimited entries like `"arduino-cli/0.21.0 ArduinoIDE/2.0.0-rc1"` if this information is
 available.
 
+### Secure Boot
+
+Some boards supports the secure boot. Basically the compiled sketch can be signed and encrypted with a [tool](#tools)
+before being flashed to the target board. The bootloader of the board is then responsible for starting the compiled
+sketch if the matching keys are used.
+
+To be able to correctly carry out all the operations at the end of the build we can leverage the
+[post build hooks](#pre-and-post-build-hooks-since-arduino-ide-165) to sign and encrypt a binary by using
+`recipe.hooks.objcopy.postobjcopy.NUMBER.pattern` key in [`platform.txt`](#platformtxt). The security keys used are
+defined in the boards file, this way there could be different keys for different boards.
+
+```
+[...]
+## Create output secure image (bin file)
+recipe.hooks.objcopy.postobjcopy.1.pattern={build.postbuild.cmd}
+#
+# IMGTOOL
+#
+
+tools.imgtool.cmd=imgtool
+tools.imgtool.build.pattern=sign --key "{build.keys.keychain}/{build.keys.sign_key}" --encrypt "{build.keys.keychain}/{build.keys.encrypt_key}" "{build.path}/{build.project_name}.bin" "{build.path}/{build.project_name}.bin" --align {build.alignment} --max-align {build.alignment} --version {build.version} --header-size {build.header_size} --pad-header --slot-size {build.slot_size}
+[...]
+
+```
+
+By having only `tools.TOOL_NAME.cmd` and `tools.TOOL_NAME.build.pattern`, we can customize the behavior with a
+[custom board option](#custom-board-options). Then in the [`boards.txt`](#boardstxt) we can define the new option to use
+a different `postbuild.cmd`:
+
+```
+[...]
+menu.security=Security setting
+
+envie_m7.menu.security.none=None
+envie_m7.menu.security.sien=Signature + Encryption
+
+envie_m7.menu.security.sien.build.postbuild.cmd="{tools.imgtool.cmd}" {tools.imgtool.build.pattern}
+envie_m7.menu.security.none.build.postbuild.cmd="{tools.imgtool.cmd}" exit
+
+envie_m7.menu.security.sien.build.keys.type=public_keys
+envie_m7.menu.security.sien.build.keys.keychain={runtime.hardware.path}/Default_Keys
+envie_m7.menu.security.sien.build.keys.sign_key=default-signing-key.pem
+envie_m7.menu.security.sien.build.keys.encrypt_key=default-encrypt-key.pem
+[...]
+```
+
+The currently we support the secure boot only with `build.keys.type=public_keys` but in the future other ways can be
+added. The security keys can be added with:
+
+- `keys.keychain` indicates the path of the dir where to search for the custom keys to sign and encrypt a binary.
+- `keys.sign_key` indicates the name of the custom signing key to use to sign a binary during the compile process.
+- `keys.encrypt_key` indicates the name of the custom encryption key to use to encrypt a binary during the compile
+  process.
+
```

Comment:
```suggestion
### Secure Boot

Some boards supports the secure boot. Basically the compiled sketch can be signed and encrypted with a [tool](#tools)
before being flashed to the target board. The bootloader of the board is then responsible for starting the compiled
sketch if the matching keys are used.

To be able to correctly carry out all the operations at the end of the build we can leverage the
[post build hooks](#pre-and-post-build-hooks-since-arduino-ide-165) to sign and encrypt a binary by using
`recipe.hooks.objcopy.postobjcopy.NUMBER.pattern` key in [`platform.txt`](#platformtxt). The security keys used are
defined in the boards file, this way there could be different keys for different boards.

```
[...]
## Create output secure image (bin file)
recipe.hooks.objcopy.postobjcopy.1.pattern={build.postbuild.cmd}
#
# IMGTOOL
#

tools.imgtool.cmd=imgtool
tools.imgtool.build.pattern=sign --key "{build.keys.keychain}/{build.keys.sign_key}" --encrypt "{build.keys.keychain}/{build.keys.encrypt_key}" "{build.path}/{build.project_name}.bin" "{build.path}/{build.project_name}.bin" --align {build.alignment} --max-align {build.alignment} --version {build.version} --header-size {build.header_size} --pad-header --slot-size {build.slot_size}
[...]

```

By having only `tools.TOOL_NAME.cmd` and `tools.TOOL_NAME.build.pattern`, we can customize the behavior with a
[custom board option](#custom-board-options). Then in the [`boards.txt`](#boardstxt) we can define the new option to use
a different `postbuild.cmd`:

```
[...]
menu.security=Security setting

envie_m7.menu.security.none=None
envie_m7.menu.security.sien=Signature + Encryption

envie_m7.menu.security.sien.build.postbuild.cmd="{tools.imgtool.cmd}" {tools.imgtool.build.pattern}
envie_m7.menu.security.none.build.postbuild.cmd="{tools.imgtool.cmd}" exit

envie_m7.menu.security.sien.build.keys.type=public_keys
envie_m7.menu.security.sien.build.keys.keychain={runtime.hardware.path}/Default_Keys
envie_m7.menu.security.sien.build.keys.sign_key=default-signing-key.pem
envie_m7.menu.security.sien.build.keys.encrypt_key=default-encrypt-key.pem
[...]
```

The currently we support the secure boot only with `build.keys.type=public_keys` but in the future other ways can be
added. The security keys can be added with:

- `keys.keychain` indicates the path of the dir where to search for the custom keys to sign and encrypt a binary.
- `keys.sign_key` indicates the name of the custom signing key to use to sign a binary during the compile process.
- `keys.encrypt_key` indicates the name of the custom encryption key to use to encrypt a binary during the compile
  process.

```

This is really valuable content, but I don't think it is appropriate for the specification. This would be better published as a guide for one possible use case of the flags.

The specification should only document the property names and the Arduino CLI flags that set them. I think the most appropriate place for them is under the "Build Process" section:

https://github.com/arduino/arduino-cli/blob/e325b1674181e47799ebc02bb76c2cbc8d15fce3/docs/platform-specification.md#build-process

Make sure to use the correct property names:

- `build.keys.keychain`
- `build.keys.sign_key`
- `build.keys.encrypt_key`

Reply:
Got it, what do you think is the best place to put this content? `/docs/secure-boot.md`?

Reply:
I'm not sure. I have pondered this in relation to some other guide/tutorial type content we have considered, such for creating Arduino [boards platforms](https://github.com/arduino/arduino-cli/issues/847) (https://github.com/arduino/arduino-cli/issues/847) or [precompiled libraries](https://forum.arduino.cc/t/precompiling-a-big-library/626128/7).

I'm kind of on the fence about whether that type of content belongs in the Arduino CLI repository, but it is developer targeted and might benefit from being maintained alongside the code that implements those capabilities.

So I think the best would be to create a sub-section of the documentation as a container for guides, like `docs/guides/secure-boot.md`

Reply:
Thanks for the useful suggestion. I'll do it right away

Reply:
Done in https://github.com/arduino/arduino-cli/commit/974f8b597f2198809f779ce67e6a0f64da88eee0

Reply:
I think it would be helpful to add a sentence or two near the start to provide a quick explanation of the reason for an Arduino boards platform developer to add a "secure boot" capability. That could include a link to an external reference that provides more detailed information on the general subject.

Reply:
Done in ea4ba1a4f91c998d99a51e48deccac1b431803d3

Reply:
Looks good!